src/source/Rtcp/RtpRollingBuffer.c

```diff
@@ -33,9 +33,12 @@ STATUS addRtpPacket(PRollingBuffer pRollingBuffer, PRtpPacket pRtpPacket)
 {
     STATUS retStatus = STATUS_SUCCESS;
     PRtpPacket pRtpPacketCopy = NULL;
+    PBYTE rawPacketCopy = NULL;
     CHK(pRollingBuffer != NULL && pRtpPacket != NULL, STATUS_NULL_ARG);
 
-    CHK_STATUS(createRtpPacketFromBytes(pRtpPacket->pRawPacket, pRtpPacket->rawPacketLength, &pRtpPacketCopy));
+    rawPacketCopy = MEMALLOC(pRtpPacket->rawPacketLength);
+    MEMCPY(rawPacketCopy, pRtpPacket->pRawPacket, pRtpPacket->rawPacketLength);
+    CHK_STATUS(createRtpPacketFromBytes(rawPacketCopy, pRtpPacket->rawPacketLength, &pRtpPacketCopy));
```

Comment:
Not sure I understand this. Doesn't createRtpPacketFromBytes already allocate the storage for the bytes?

Reply:
createRtpPacketFromBytes will not copy the raw packet buffer, so if only calling it without copying raw packet, when we use the raw packet pointer later in the new rtp packet, it can be freed somewhere else.